### ISSUE TITLE
Android Build Error - [string/google_app_id] - Duplicate Resource

### ISSUE THREAD

**Issue by emyymk** [chemerisuk/cordova-plugin-firebase-analytics#37]
Hi, using
cordova 7.0.1
android 6.2.3

Android build failed message

```
Execution failed for task ':mergeDebugResources'.
[string/google_app_id] .../platforms/android/res/values/strings.xml	
[string/google_app_id] .../platforms/android/build/generated/res/google-services/debug/values/values.xml: Error: Duplicate resources
```

- Removed the google_app_id from platforms/android/res/values/strings.xml	
- Same problem
- Removed the google_app_id from platforms/android/build/generated/res/google-services/debug/values/values.xml
- Same problem
- Removed google_app_id from both files
- Same problem

Any idea?

**Comment by chemerisuk** [comment]
Read your logs carefully. I believe you have a dup string in resources, may be other than google_app_id

**Comment by emyymk** [comment]
Thank you for your response, chemerisuk.

Well. I started all again, because I was using an old version of plugins. 

And now I have this msg error similar to [this issue](https://github.com/chemerisuk/cordova-plugin-firebase-analytics/issues/33), but i am using [phonegap-plugin-push](https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/INSTALLATION.md#co-existing-with-plugins-that-use-firebase).
```
What went wrong:
A problem occurred configuring root project 'android'.
Cannot add task ':processDebugGoogleServices' as a task with that name already exists.
```

After dealing with some msg erros, my bundle.graddle is now:
```
buildscript {
    repositories {
           jcenter()
           mavenLocal()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:2.2.3'
        classpath 'com.google.gms:google-services:3.0.0'
    }
}

dependencies {
    compile 'com.google.android.gms:play-services:11.0.+'
    compile 'com.google.android.gms:play-services-gcm:11.0.+'
}

apply plugin: 'com.google.gms.google-services'
```

My project.properties

```
target=android-25
android.library.reference.1=CordovaLib
cordova.system.library.1=com.google.firebase:firebase-core:11.0.+
cordova.gradle.include.1=cordova-plugin-firebase-analytics/coop-build.gradle
cordova.gradle.include.2=phonegap-plugin-push/coop-push.gradle
cordova.system.library.2=com.android.support:support-v13:23+
cordova.system.library.3=com.google.android.gms:play-services-gcm:11.0.+
cordova.system.library.4=me.leolin:ShortcutBadger:1.1.14@aar
```

- I have already remove/add Android platform
- I have tried with  'com.google.gms:google-services:3.0.0' and  'com.google.gms:google-services:3.1.0' 

Please, do you know how can i solve this?

**Comment by chemerisuk** [comment]
This is a trash from a some other plugin. It will be ok, because this constant value is generated from your 'google-services.json' now

**Comment by nickforddesign** [comment]
Fair enough, it does appear to be working without issue. Thanks!

**Comment by ravikumarthuta** [comment]
now i am facing same problam.what is the solution please let me know.

**Comment by FavoriteFave** [comment]
@nickforddesign this worked for me. thank you!

**Comment by eswarn24** [comment]
@nickforddesign Not sure, but the steps you provided worked for me. 

I was working in a project to completely upgrade a cordova project & plugins to latest version

In my case, I was using Push services in GCM and used to build the project directly in android studio rather than cordova build (Though not recommended). 

Since google-service.json was introduce in firebase rather than GCM, it is mandatory for latest push plugin in cordova to build android projects.

So, I migrated the project to firebase and downloaded latest google-service.json from console portal. I included it in root directory of cordova project and during build process I encountered this error.

Your solution saved my time rather than banging my head on this issue

Thank you!!
